### PR TITLE
Environment file should be a simple key: value pair file.

### PR DESCRIPTION
**Pull request by mhahn**
We no longer want to double yaml parse the environment file. This treats
the environment as a group of dumb key: value pairs, where the raw
string value will be passed to the yaml config.

### CODE REVIEW

File: stacker/commands/stacker/base.py [hunk 1]
```diff
@@ -149,11 +149,12 @@ def add_arguments(self, parser):
         parser.add_argument("-v", "--verbose", action="count", default=0,
                             help="Increase output verbosity. May be specified "
                                  "up to twice.")
-        parser.add_argument('environment', type=yaml_file_type,
+        parser.add_argument('environment', type=environment_file,
                             default={},
-                            help="Path to a yaml environment file. The values in "
-                                 "the environment file can be used in the stack "
-                                 "config as if it were a string.Template type: "
+                            help="Path to a simple key/value pair environment "
```
Comment (phobologic) [inline stacker/commands/stacker/base.py]: `key: value` maybe better?

File: stacker/environment.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
+
+def parse_environment(raw_environment):
+    environment = {}
+    for line in raw_environment.split('\n'):
```
Comment (phobologic) [inline stacker/environment.py]: Should probably handle `#comments` as well

File: stacker/environment.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+        if ':' not in line:
```
Comment (phobologic) [inline stacker/environment.py]: Hmm - this is a little too loose.  You're trying to catch blank lines, right?

I'd maybe go with:

```
if not line:
    continue

try:
    key, value = line.split(":", 1)
except ValueError:
    raise Exception("Environment must be in key: value format")
```

File: stacker/tests/test_environment.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import unittest
+
+from stacker.environment import parse_environment
+
+test_env = """key1: value1
```
Comment (phobologic) [inline stacker/tests/test_environment.py]: Per above - add blank lines & comments just to make sure we don't miss anything.

**Comment by mhahn** [comment]
@phobologic comments addressed

**Pull request merged**